src/main/scala/com/fulcrumgenomics/bam/ClipBam.scala

```diff
@@ -102,6 +106,8 @@ class ClipBam
         case _ => Unit
       }
 
+      if (this.convertSoftClipToHardClip) template.allReads.foreach { r => this.clipper.toHardClip(r) }
```

Comment:
So I'm pretty sure this does nothing since in `SamRecordClipper` you return immediately if mode != Hard, and you require above that mode == Soft.

Reply:
This would also argue for needing a test for the changes to ClipBam.